tests/integration/pilot/analysis/main_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analysis
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/framework/resource/environment"
+)
+
+var (
+	i istio.Instance
+	p pilot.Instance
+)
+
+// TestMain defines the entrypoint for pilot tests using a standard Istio installation.
+// If a test requires a custom install it should go into its own package, otherwise it should go
+// here to reuse a single install across tests.
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("pilot_analysis_test", m).
+		RequireSingleCluster().
+		RequireEnvironment(environment.Kube).
+		SetupOnEnv(environment.Kube, istio.Setup(&i, nil)).
+		SetupOnEnv(environment.Kube, istio.Setup(&i, func(cfg *istio.Config) {
+			cfg.ControlPlaneValues = `
+components:
+  base:
+    enabled: false
+  pilot:
+    enabled: true
+  ingressGateways:
+addonComponents:
+  prometheus:
+    enabled: false
+revision: canary
+values:
+  global:
+    istiod:
+      enableAnalysis: true
```

Comment:
The addition of these lines to values seems to cause pilot to be installed twice, which causes problems.  Does anyone know why this is happening?

Reply:
You are calling `istio.Setup` twice, which installs it twice. Copying from the pilot main test was a bad example because its doing something it shouldn't (changed in https://github.com/istio/istio/pull/22336).

Reply:
Thank you for clarifying!